### PR TITLE
feat(mfa): wrap change primary email action with MFA guard

### PR DESCRIPTION
**Pull request by MagentaManifold**
## Because

- we want mfa for sensitive actions

## This pull request

- wraps change primary email action with MFA guard

## Issue that this pull request solves

Closes: FXA-12226

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [x] I have added necessary documentation (if appropriate).
- [ ] I have verified that my changes render correctly in RTL (if appropriate).

## Screenshots (Optional)

Please attach the screenshots of the changes made in case of change in user interface.

## Other information (Optional)

Any other information that is important to this pull request.

### CODE REVIEW

File: packages/fxa-settings/src/models/Account.ts [hunk 1]
```diff
@@ -1239,6 +1239,49 @@
...
+        avatar: (existing, { readField }) => {
+          const id = readField<string>('id', existing);
+          const oldUrl = readField<string>('url', existing);
+          return getNextAvatar(id, oldUrl, email, this.displayName);
+        },
+      },
+    });
+
+    const legacyLocalStorageAccount = currentAccount()!;
+    legacyLocalStorageAccount.email = email;
```
[the remark this answers is not in the document]
Comment (MagentaManifold) [reply packages/fxa-settings/src/models/Account.ts:1279]: I don't get it

File: packages/fxa-auth-server/lib/routes/emails.js [hunk 2]
```diff
@@ -1008,6 +1107,28 @@ module.exports = (
...
+          payload: isA.object({
+            email: validators
+              .email()
+              .required()
+              .description(DESCRIPTION.emailNewPrimary),
+          }),
+        },
+        response: {},
+      },
+      handler: handlers.setPrimaryEmailPost,
```
Comment (dschom) [inline packages/fxa-auth-server/lib/routes/emails.js]: Can we reuse existing handler, `POST recovery_email/set_primary`. Here is an example https://github.com/mozilla/fxa/pull/19444#discussion_r2349755280. This won't disrupt the commit history, is fewer changes, and avoids copy and pasting code.

File: packages/fxa-auth-server/lib/routes/emails.js [hunk 3]
```diff
@@ -1008,6 +1107,28 @@ module.exports = (
         return {};
       },
     },
+    {
+      method: 'POST',
+      path: '/mfa/recovery_email/set_primary',
+      options: {
+        ...EMAILS_DOCS.RECOVERY_EMAIL_SET_PRIMARY_POST,
```
Comment (dschom) [inline packages/fxa-auth-server/lib/routes/emails.js]: Should be, `EMAILS_DOCS.MFA_RECOVERY_EMAIL_SET_PRIMARY_POST`

File: packages/fxa-settings/src/components/Settings/UnitRowSecondaryEmail/index.test.tsx [hunk 4]
```diff
@@ -313,38 +345,51 @@ describe('UnitRowSecondaryEmail', () => {
...
-      await act(async () => {
-        fireEvent.click(screen.getByTestId('secondary-email-make-primary'));
+      await user.click(screen.getByTestId('secondary-email-make-primary'));
+      await waitFor(() => {
+        expect(settingsContext.alertBarInfo?.error).toHaveBeenCalledTimes(1);
       });
-      expect(settingsContext.alertBarInfo?.error).toHaveBeenCalledTimes(1);
+    });
+
+    it('displays the MFA guard if the JWT is invalid', async () => {
```
Comment (dschom) [inline packages/fxa-settings/src/components/Settings/UnitRowSecondaryEmail/index.test.tsx]: Nit: I think we already have tests like this on the Guard itself. It seems kind of like we are retesting guard functionality with test. Let me know if this is covering an edge case we can't test in `MfaGuard/index.test.tsx`.

File: packages/fxa-settings/src/components/Settings/UnitRowSecondaryEmail/index.tsx [hunk 5]
```diff
@@ -58,29 +67,51 @@ export const UnitRowSecondaryEmail = () => {
...
+            l10n.getString(
+              'se-set-primary-successful-2',
+              { email },
+              `${email} is now your primary email`
+            )
+          );
+          setPendingChangePrimary(undefined);
+        } catch (e) {
+          if (isInvalidJwtError(e)) {
+            clearMfaAndJwtCacheOnInvalidJwt(e, 'email');
```
Comment (dschom) [inline packages/fxa-settings/src/components/Settings/UnitRowSecondaryEmail/index.tsx]: Is there where we ran into a problem with the error boundary?

File: packages/fxa-settings/src/components/Settings/UnitRowSecondaryEmail/index.tsx [hunk 6]
```diff
@@ -58,29 +67,51 @@ export const UnitRowSecondaryEmail = () => {
...
+              null,
+              'Sorry, there was a problem changing your primary email'
+            )
+          );
+          setPendingChangePrimary(undefined);
+        }
+      },
+      [account, alertBar, l10n]
+    );
+    useEffect(() => {
```
Comment (dschom) [inline packages/fxa-settings/src/components/Settings/UnitRowSecondaryEmail/index.tsx]: Nit: Does this fix a re-render problem? If so, put a comment.

File: packages/fxa-settings/src/components/Settings/UnitRowSecondaryEmail/index.tsx (hunk 5 above)
Comment (nshirley) [reply packages/fxa-settings/src/components/Settings/UnitRowSecondaryEmail/index.tsx]: @dschom - this is identical to how I [had it setup](https://github.com/mozilla/fxa/pull/19453/files#diff-a1a866e1db1cbbde7cfda24e1a2186e231ebf9c42d1fcdaebcbb78cfcf1522b8R102). Because of the odd use case and the fact that the mfaguard is no longer mounted by this point we had to manually handle clearing the cache which then is picked up and re-triggers the guard.

File: packages/fxa-settings/src/models/Account.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (dschom) [reply packages/fxa-settings/src/models/Account.ts:1279]: Hmm, I've filed a [follow up](FXA-12450) for this. Let's not let it block this PR...

File: packages/fxa-settings/src/components/Settings/UnitRowSecondaryEmail/index.test.tsx (hunk 4 above)
Comment (MagentaManifold) [reply packages/fxa-settings/src/components/Settings/UnitRowSecondaryEmail/index.test.tsx]: Same as Nick's PR, I need this because there is a workaround in this component so extra testing is needed to make sure it works.
